packages/node/src/node.ts

```diff
@@ -241,25 +241,34 @@ export class Node {
             params.multisigAddress,
             context.finalCommitment!
           );
-        } else {
-          if (!context.commitments[0]) {
-            throw new Error(
-              `State transition without commitment: ${JSON.stringify(message)}`
+        } else if (
+          protocol === Protocol.Install ||
+          protocol === Protocol.Uninstall ||
+          protocol === Protocol.Update
+        ) {
+          /// install, update, uninstall,
+          /// All other protocols are stored under the key
+          /// appIdentityHash/protocolName. This means that an app only has one
+          /// update commitment.
+          const params = message.params as any;
+          if (!params.multisigAddress) {
+            throw Error(
+              "WRITE_COMMITMENT: params did not contain multisigAddress"
             );
           }
-          const transaction = context.commitments[0].transaction([
-            context.signatures[0] // TODO: add counterparty signature
-          ]);
-          if (!context.appIdentityHash) {
-            throw new Error(
-              `appIdentityHash required to store commitment. protocol=${protocol}`
+          if (!getAddress(params.multisigAddress)) {
```

Comment:
Why is this here? Surely it would have been caught well ahead of time in the instruction executor process